src/Corax/Analyzers/Analyzer.cs

```diff
@@ -0,0 +1,281 @@
+﻿using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using Corax.Pipeline;
+
+namespace Corax
+{
+    public unsafe class Analyzer : IDisposable
+    {
+        public static readonly ArrayPool<byte> BufferPool = ArrayPool<byte>.Create();
+        public static readonly ArrayPool<Token> TokensPool = ArrayPool<Token>.Create();
+
+        private readonly delegate*<Analyzer, ReadOnlySpan<byte>, ref Span<byte>, ref Span<Token>, void> _func;
+        private readonly Analyzer _inner;
+        private readonly ITokenizer _tokenizer;
+        private readonly ITransformer[] _transformers;
+        private readonly float _sourceBufferMultiplier;
+        private readonly float _tokenBufferMultiplier;
```

Comment:
Add a comment explaining what those are for.

Reply:
Also, wouldn't it be better to make them `int`, even if we use a bit more?

Reply:
The idea is that we can say, we dont need the full buffer just half of it or something like that. We can use `int` but that would mean that we cannot ask for an smaller buffer even if we know that we are gonna kill half of the data.

Reply:
Doubling the size (and power of 2) would likely be sufficient. There is just one of those at any given time, anyway

Reply:
That's not how that works. That is to actually calculate on construction the expected size, then we will rent the buffers which will guarantee from the way `ArrayPool` works to give us power of 2 buffers.